sources/packages/api/src/route-controllers/institution/models/institution.dto.ts

```diff
@@ -74,32 +65,51 @@ export class InstitutionDto extends PartialType(CreateInstitutionDto) {
   @IsOptional()
   institutionTypeName: string;
 }
+export class InstitutionAddress {
+  addressLine1: string;
+  addressLine2: string;
+  city: string;
+  provinceState: string;
+  country: string;
+  postalCode: string;
+}
 
-export class AESTInstitutionDetailDto {
-  legalOperatingName: string;
+export class InstitutionContactDto {
+  @Allow()
+  primaryContactEmail: string;
+  @Allow()
+  primaryContactFirstName: string;
+  @Allow()
+  primaryContactLastName: string;
+  @Allow()
+  primaryContactPhone: string;
+  @Allow()
+  mailingAddress: InstitutionAddress;
+}
+
+export class InstitutionProfileDto extends InstitutionContactDto {
+  @Allow()
   operatingName: string;
+  @Allow()
   primaryPhone: string;
+  @Allow()
   primaryEmail: string;
+  @Allow()
   website: string;
+  @Allow()
   regulatingBody: string;
-  institutionTypeName: string;
-  formattedEstablishedDate: string;
-  primaryContactEmail: string;
-  primaryContactFirstName: string;
-  primaryContactLastName: string;
-  primaryContactPhone: string;
-  legalAuthorityEmail: string;
-  legalAuthorityFirstName: string;
-  legalAuthorityLastName: string;
-  legalAuthorityPhone: string;
-  address: {
-    addressLine1: string;
-    addressLine2: string;
-    city: string;
-    country: string;
-    provinceState: string;
-    postalCode: string;
-  };
+  @Allow()
+  establishedDate: Date;
+  @Allow()
+  institutionType: number;
+}
+
+export class InstitutionReadOnlyDto extends InstitutionProfileDto {
```

Comment:
Does _ReadOnly_ here mean that the information belongs to a UI that will be read-only or that the DTO is not suitable to be edited?

Reply:
I meant that is readOnly dto is for common GET API.

Reply:
DTO name changed.